signing/src/test/java/tech/pegasys/web3signer/signing/config/SignerLoaderEnhancedTest.java

```diff
@@ -0,0 +1,486 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.signing.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.spy;
+
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.web3signer.BLSTestUtil;
+import tech.pegasys.web3signer.common.Web3SignerMetricCategory;
+import tech.pegasys.web3signer.keystorage.aws.AwsSecretsManagerProvider;
+import tech.pegasys.web3signer.keystorage.common.MappedResults;
+import tech.pegasys.web3signer.keystorage.hashicorp.HashicorpConnectionFactory;
+import tech.pegasys.web3signer.signing.ArtifactSigner;
+import tech.pegasys.web3signer.signing.BlsArtifactSigner;
+import tech.pegasys.web3signer.signing.config.metadata.BlsArtifactSignerFactory;
+import tech.pegasys.web3signer.signing.config.metadata.SigningMetadataException;
+import tech.pegasys.web3signer.signing.config.metadata.parser.SignerParser;
+import tech.pegasys.web3signer.signing.config.metadata.parser.YamlMapperFactory;
+import tech.pegasys.web3signer.signing.config.metadata.parser.YamlSignerParser;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import de.neuland.assertj.logging.ExpectedLogging;
+import de.neuland.assertj.logging.ExpectedLoggingAssertions;
+import de.neuland.assertj.logging.LogEvent;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * Enhanced test suite for SignerLoader focusing on: - Timeout behavior - Batch size processing -
+ * Sequential vs parallel processing
+ */
+@ExtendWith(MockitoExtension.class)
+public class SignerLoaderEnhancedTest {
```

Comment:
The tests separated from the `SignerLoaderTest` seems a bit arbitrary. Can this be renamed to to something that better describes what these tests are or just include these in the main `SignerLoaderTest`

Reply:
Moved all the tests under `SignerLoaderTest` class.